### PR TITLE
feat: record game hash id for sessions, unlocks, and lb entries

### PR DESCRIPTION
**Pull request by wescopeland**
This PR updates several Connect API routes & associated actions/helpers to ensure that the `m` or `x` param sent by `rc_client` for the game hash MD5 is paired to the corresponding game hash in the database and recorded as the `game_hash_id` value.

The following tables will now have their `game_hash_id` values correctly populated:
* `player_sessions`
* `player_games`

Additionally, the following tables will now have their `player_session_id` values correctly populated:
* `player_achievements`
* `leaderboard_entries`

With these values being recorded, we can begin to build insights/analytics into hash use for sets.

### CODE REVIEW

File: app/Platform/Actions/ResumePlayerSession.php [hunk 1]
```diff
@@ -65,6 +65,10 @@ public function execute(
             }
             $playerSession->rich_presence_updated_at = $timestamp > $playerSession->rich_presence_updated_at ? $timestamp : $playerSession->rich_presence_updated_at;
 
+            if ($gameHash && !$playerSession->game_hash_id) {
+                $playerSession->game_hash_id = $gameHash->id;
+            }
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply app/Platform/Actions/ResumePlayerSession.php:71]: Updated in latest.

[the remark this answers is not in the document]
Comment (Jamiras) [reply app/Platform/Actions/ResumePlayerSession.php:71]: I think we want to conditionally filter on `game_hash_id` (similar to how user agent is handled). If a request comes through without a hash, but other had hashes, we'd still want to associate them together. We can only reasonably say they're from separate sessions if they have unique non-blank hashes.

[the remark this answers is not in the document]
Comment (wescopeland) [reply app/Platform/Actions/ResumePlayerSession.php:76]: Ah, that makes sense. Addressed in latest.

File: public/dorequest.php [hunk 2]
```diff
@@ -306,7 +308,12 @@ function DoRequestError(string $error, ?int $status = 200, ?string $code = null)
                 $activityMessage = utf8_sanitize($activityMessage);
             }
 
-            PlayerSessionHeartbeat::dispatch($user, $game, $activityMessage);
+            $gameHashMd5 = request()->post('x') ?? request()->query('x') ?? null;
```
Comment (Jamiras) [inline public/dorequest.php:311]: Isn't this just:
```
$gameHashMd5 = request()->input('x');
```

Comment (wescopeland) [reply public/dorequest.php]: Yep. I managed to forget that `request()->input()` was a thing. Updated in latest.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**